streams/src/test/java/org/apache/kafka/streams/processor/internals/StoreChangelogReaderTest.java

```diff
@@ -385,6 +385,58 @@ public void shouldThrowTaskMigratedExceptionIfEndOffsetGetsExceededDuringRestore
         } catch (final TaskMigratedException expected) { /* ignore */ }
     }
 
+    @Test
+    public void shouldThrowTaskMigratedExceptionIfEndOffsetGetsExceededDuringRestoreForChangelogTopicAfterComplete() {
+        final int messages = 10;
+        setupConsumer(messages, topicPartition);
+        consumer.simulateEndOffsetsUpdatedByAnotherThread(Collections.singletonMap(topicPartition, 12L));
+        changelogReader.register(new StateRestorer(topicPartition, restoreListener, null, Long.MAX_VALUE, true,
+                                                   "storeName"));
+        expect(active.restoringTaskFor(topicPartition)).andReturn(task);
+        replay(active);
+
+        try {
+            changelogReader.restore(active);
+            fail("Should have thrown TaskMigratedException");
+        } catch (final TaskMigratedException expected) { /* ignore */ }
+    }
+
+    @Test
+    public void shouldNotThrowTaskMigratedExceptionIfEndOffsetIsUpdatedDuringRestoreForSourceTopicAfterComplete() {
+        final int messages = 10;
+        setupConsumer(messages, topicPartition);
+        consumer.simulateEndOffsetsUpdatedByAnotherThread(Collections.singletonMap(topicPartition, 11L));
+        changelogReader.register(new StateRestorer(topicPartition, restoreListener, null, 10, true,
+                                                   "storeName"));
+
+        expect(active.restoringTaskFor(topicPartition)).andReturn(task);
+        replay(active);
+
+        try {
+            changelogReader.restore(active);
+        } catch (final TaskMigratedException expected) {
```

Comment:
we remove `try-catch-fail` here -- if there is an exception the test won't pass anyway. (same below)

Reply:
ack